SUMMARY.md

```diff
@@ -0,0 +1,32 @@
+# Summary
+
+- [Introduction](./intro.md)
+- [Idioms](./idioms/README.md)
+	* [Concatenating Strings with `format!`](./idioms/concat-format.md)
+	* [Constructor](./idioms/ctor.md)
+	* [The `Default` Trait](./idioms/default.md)
+	* [Collections Are Smart Pointers](./idioms/deref.md)
+	* [Finalisation in Destructors](./idioms/dtor-finally.md)
+	* [`mem::replace(_)`](./idioms/mem-replace.md)
+	* [On-Stack Dynamic Dispatch](./idioms/on-stack-dyn-dispatch.md)
+	* [Iterating over an `Option`](./idioms/option-iter.md)
+	* [Pass Variables to Closure](./idioms/pass-var-to-closure.md)
+	* [Privacy For Extensibility](./idioms/priv-extend.md)
+	* [Easy doc initialization](./idioms/rustdoc-init.md)
+	* [Temporary mutability](./idioms/temporary-mutability.md)
+	
+- [Design Patterns](./patterns/README.md)
```

Comment:
Here as well, some subcategories might be nice?

Reply:
Categories could be:
- Creational Patterns
- Structural Patterns
- Behavioral Patterns

Reply:
Can't review this know, since I haven't read all the patterns.
Feel free to add them if you think they are right. Somebody else will criticize them if they think they are wrong (I hope)

Reply:
Probably would do that in another PR, because it would make sense to also structure the repository like this I feel with subfolders? And that would be a bit of a change that needs to be discussed separately

Reply:
With subfolders I mean `patterns/structural`, `patterns/bahavioral` and `patterns/creational` as subfolders in `patterns`. Referring to the structure in `Design patterns` (by Gamma, Helm, Johnson, Vlissides).

Reply:
yes, but we should do it once we are sure we closed/merged all the PRs, otherwise we will have a lot of merge conflicts.

Reply:
true, will open an issue to track